lib/Service/MailManager.php

```diff
@@ -321,17 +321,43 @@ public function moveMessage(Account $sourceAccount,
 	 */
 	public function deleteMessage(Account $account,
 		string $mailboxId,
-		int $messageId): void {
-		$this->eventDispatcher->dispatch(
-			BeforeMessageDeletedEvent::class,
-			new BeforeMessageDeletedEvent($account, $mailboxId, $messageId)
-		);
-
+		int $messageUid): void {
 		try {
 			$sourceMailbox = $this->mailboxMapper->find($account, $mailboxId);
 		} catch (DoesNotExistException $e) {
 			throw new ServiceException("Source mailbox $mailboxId does not exist", 0, $e);
 		}
+
+		$client = $this->imapClientFactory->getClient($account);
+		try {
+			$this->deleteMessageWithClient($account, $sourceMailbox, $messageUid, $client);
+		} finally {
+			$client->logout();
+		}
+	}
+
+	/**
+	 * @param Account $account
+	 * @param Mailbox $mailbox
+	 * @param int $messageId
+	 * @param Horde_Imap_Client_Socket $client
+	 *
+	 * @throws ServiceException
+	 * @throws ClientException
+	 * @throws TrashMailboxNotSetException
+	 *
+	 * @todo evaluate if we should sync mailboxes first
+	 */
+	public function deleteMessageWithClient(
```

Comment:
Good idea. 

I ran into a similar issue with flagMessage. It's a bit sad that most methods only accept one uid. 

Especially your case would benefit from a deleteMessages method because imap also accepts sending more than one uid with move or expunge. 

That would need some testing, especially with the multiple events dispatched.